### ISSUE TITLE
TODO: incorporate helpful error message for getLOESSfit

### ISSUE THREAD

**Issue by singjc** [shubham1637/DIAlignR#20]
`getLOESSfit` returns an uninformative error when using non-optimal *maxGlobalFdr* for loess fit.

```> fit <- DIAlignR:::getLOESSfit(features, ref, eXp, 0.05, 0.1)
Error in vecseq(f__, len__, if (allow.cartesian || notjoin || !anyDuplicated(f__,  : 
  Join results in 5359 rows; more than 5228 = nrow(x)+nrow(i). Check for duplicate key values in i each of which join to the same group in x over and over again. If that's ok, try by=.EACHI to run j for each group to avoid the large allocation. If you are sure you wish to proceed, rerun with allow.cartesian=TRUE. Otherwise, please search for this error message in the FAQ, Wiki, Stack Overflow and data.table issue tracker for advice.
```

There is a check that increments the *maxGlobalFdr* by 10 times [get_global_fit.R#L246](https://github.com/shubham1637/DIAlignR/blob/f028452d38ea9cedddf917d782dc0ca3ddc449be/R/get_global_fit.R#L246), but in my case I had to decrease the *FDR* to 0.01 for one run pair, and even further 0.006 for another run pair. Might need to implement another check that decrements *maxGlobalFdr*.

**Comment by shubham1637** [comment]
You may provide a lower fdr value to startwith `params[["globalAlignmentFdr"]]`

**Comment by singjc** [comment]
> You may provide a lower fdr value to startwith `params[["globalAlignmentFdr"]]`

Yeah that's what I did, but I had to debug the code line by line to figure out what parameter I had to change. The internal error complains about a table join, so it wasn't as straight forward as what caused the error. Just leaving this issue as a self-reminder to put a more helpful error message, to direct the user what parameter they should adjust. I can add a message when I make an update.
